icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -350,36 +537,97 @@ void opana::ICARUSFlashAssAna::analyze(art::Event const& e) {
   if( !fTriggerLabel.empty() ) { 
 
+      // Beam information
       art::Handle<std::vector<sim::BeamGateInfo>> beamgate_handle;
       e.getByLabel( fTriggerLabel, beamgate_handle );
-
+      
       if( beamgate_handle.isValid() ) {
 
         for( auto const & beamgate : *beamgate_handle ) {
 
           m_beam_gate_start = beamgate.Start(); 
           m_beam_gate_width = beamgate.Width(); 
-          m_beam_type = beamgate.BeamType() ;
+          m_beam_type = beamgate.BeamType() ; 
 
         }
 
       }
 
       else {
-        std::cout << "Invalid Trigger Data product " << fTriggerLabel.label() << "\n" ;
+        if( fDebug) { std::cout << "No sim::BeamGateInfo associated to label: " << fTriggerLabel.label() << "\n" ; }
+      }
+
+      // Now trigger information
+      art::Handle<sbn::ExtraTriggerInfo> trigger_handle;
+      e.getByLabel( fTriggerLabel, trigger_handle );
+
+      if( trigger_handle.isValid() ) {
+
+        sbn::triggerSource bit = trigger_handle->sourceType;
+
+        m_gate_type = (unsigned int)bit; 
+        m_gate_name = bitName(bit);
+        m_trigger_timestamp = trigger_handle->triggerTimestamp;
+        m_gate_start_timestamp =  trigger_handle->beamGateTimestamp;
+        m_trigger_gate_diff = m_trigger_timestamp-m_gate_start_timestamp;
+
+        std::cout << m_trigger_timestamp << "; " << m_gate_start_timestamp << " ;" << m_trigger_gate_diff << std::endl;
+
+      }
+      else{
+        if( fDebug ){ std::cout << "No raw::Trigger associated to label: " << fTriggerLabel.label() << "\n" ; }
       }
 
   }
 
   else {
-     std::cout << "Trigger Data product " << fTriggerLabel.label() << " not found!\n" ;
+     if( fDebug ){ std::cout << "Trigger Data product " << fTriggerLabel.label() << " not found!\n" ; }
   }
 
 
-  // Now we take care of the labels
+  /* 
+  Now we work on the waveforms if we are allowed to
+  */
+
+    if( !fOpDetWaveformLabels.empty() && fSaveWaveformInfo ) { 
+    
+      for( size_t i=0; i<fOpDetWaveformLabels.size(); i++ ) {
+
+        auto const label = fOpDetWaveformLabels[i];
+
+        art::Handle<std::vector<raw::OpDetWaveform>> wfm_handle;
+        e.getByLabel( label, wfm_handle );
+
+        if( wfm_handle.isValid() && !wfm_handle->empty() ) {
+
+          for( auto const & wave : *wfm_handle ){
+
+            m_channel_id = wave.ChannelNumber();
+            m_nticks = wave.Waveform().size();
+            m_baseline = Median( wave.Waveform() ); 
+            m_chargesum = std::accumulate( wave.Waveform().begin(), wave.Waveform().end(), 0, 
+             [ & ](short x, short y){ return ((m_baseline-x) + (m_baseline-y)) ; } );
+
+            fOpDetWaveformTrees[i]->Fill();
+            
+          }
+        }
+      }
+    }
+
+
+  /*
+    Now we take care of the flashes: we separate the case where we have a flash and the case where we have not a flash
+  */
+
   if ( !fFlashLabels.empty() ) {
 
     for ( size_t i=0; i<fFlashLabels.size(); i++  ) {
```

Comment:
Same comment about the name of the index `i` as above (although you inherited this).